custom/inddex/food.py

```diff
@@ -234,6 +233,18 @@ def recipe_id(self):
             return self.caseid
         return self.recipe_case_id or self.already_reported_recipe_case_id or 'NO_RECIPE'
 
+    _total_grams = 'NOT_YET_COMPUTED'
```

Comment:
I may be mutating a class attribute here, but I'm not gonna be happy about it.  This approach should prevent me from accidentally writing something which depends on `total_grams` when it hasn't yet been calculated.  I briefly considered doing something similar for `recipe_num_ingredients` and `ingr_recipe_total_grams_consumed`, but figured that'd be overkill, since they aren't used in calculations. :crossed_fingers:

Reply:
Wouldn't it be better to put `self._total_grams = 'NOT_YET_COMPUTED'` in `__init__()`? It wouldn't be *right next to* the property, but it might make you happier :sweat_smile: and it would achieve the same goal.

Reply:
The thing I'm not happy about is that `total_grams` needs to be initialized - calling it may or may not work, depending on how the class has been handled up to that point.  The instance state is mutated outside of the constructor.

Reply:
Oh, I definitely agree, having to mutate the instance after you start using it in order for it to work properly, is not great!

But there is more. Something triggers my OCD by setting its initial value as a class attribute, and then setting its value later on the instance. What you've done here is fine because the class attribute's value is immutable. But if it was mutable, it's possible to change the class instead of the instance:
```python
In [29]: class Foo:
    ...:     _bar = ['SPAM']
    ...:     @property
    ...:     def bar(self):
    ...:         if self._bar == ['SPAM']:
    ...:             raise AttributeError('Not just SPAM')
    ...:         return self._bar
    ...:     @bar.setter
    ...:     def bar(self, value):
    ...:         self._bar.append(value)
    ...:         

In [30]: foo1 = Foo()

In [31]: foo1.bar
    ...
AttributeError: Not just SPAM

In [32]: foo1.bar = 'eggs'

In [33]: foo1.bar
Out[33]: ['SPAM', 'eggs']

In [34]: foo2 = Foo()

In [35]: foo2.bar
Out[35]: ['SPAM', 'eggs']
```
Setting its initial value in the constructor makes that impossible. It doesn't fix the unhappiness. It just makes the sadness less awkward, at least for me and my OCD. :upside_down_face:

Reply:
Looking at this again, I'm not sure my approach (or this one) offers any advantage over simply not setting the variable at all.  I want an error to occur if someone attempts to access `row.total_grams` before it's defined, but that's also how that _already works_.  Ie
```python
In [1]: class Row:
   ...:     pass
   ...: 

In [2]: row = Row()

In [3]: row.total_grams
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-899ad55a5c32> in <module>()
----> 1 row.total_grams

AttributeError: 'Row' object has no attribute 'total_grams'

In [4]: row.total_grams = 3

In [5]: row.total_grams
Out[5]: 3
```
I suppose I could also put something in `__getattr__` to raise a more helpful `AttributeError`.

thoughts?

Reply:
Lol, actually, I really like that plan!

Considering `__getattr__` already exists, that might be nice. Otherwise I'd suggest adding a comment in `__init__()` to explain that `total_grams` is a thing and to expect it to be set later.